### PR TITLE
[IMP] Factorizing hr_cae

### PR DESCRIPTION
**Pull request by quentinDupont**
Hello,
I think that `hr_cae` depends on too many modules + his area is too wide and it would be great to make it more modular.
Here are the changes I propose to make `hr_cae` more modular

**Four new modules for the moment**
`l10n_fr_hr` that could be suggested to OCA https://github.com/OCA/l10n-france
`hr_cae_recruitment` that take everything linked to recruitment in employee and applicants
`hr_cae_bank` for bank related fields
`hr_cae_contribution` for handling contributions

**Removed**
Remove dependency `base_location_geonames_import` that seems to be an autonomous module not mandatory for `hr_cae` 

**To discuss**
For `hr_cae_bank`, it depends on `account_banking_sepa_direct_debit` but at the moment we only use field `valid_mandate_id` which is used on `account_banking_mandate`, can we change the dependency ?
Are modules's name ok ?

### CODE REVIEW

File: l10n_fr_hr/__manifest__.py
Comment (legalsylvain) [inline l10n_fr_hr/__manifest__.py]: GRAO ;-)

File: l10n_fr_hr/__manifest__.py
Comment (legalsylvain) [inline l10n_fr_hr/__manifest__.py]: Regarding this module, I think that :
- the name should be l10n_fr_department_hr_employee (as it will come in to l10n-france, all the module should be prefixed by l0n_fr).
- should not depend on ``hr_cae``, but depend only on ``hr_employee``. 
1) I don't see what it is required in ``hr_cae``
2) this module will not be accepted, in OCA, if depends on non OCA modules.
- auto_install should be False, because maybe some user doens't want to see this module installed, if they don't requires there fields. ``auto_install = True`` is generally only set if module A and module B doesn't works and need a fix if installed together. (or if there is no reason to not install the glue module, if A & B are installed).

**Review by legalsylvain — COMMENTED** [review]
Partial review. (only on department module)
I'll finish the next week.

Generally, I think that the name should not be ``cae`` (or depends on ``hr_cae``, if it has nothing to do with CAE. For exemple for the country department, it is a generic module. But I think that for other modules in this PR, the same refactoring could be done. For me the ``cae`` should be used only for features that are specific to CAE.

Thanks a lot for this huge work ! 

Once finish, things will be clearer.

File: l10n_fr_hr/__manifest__.py
Comment (quentinDupont) [reply l10n_fr_hr/__manifest__.py]: Ok I missunderstand glue modules.
Thanks for explanation :ok_hand:

File: l10n_fr_hr/__manifest__.py
Comment (quentinDupont) [reply l10n_fr_hr/__manifest__.py]: I fork GRAP to GRAO didn't I tell you ? :stuck_out_tongue:

File: l10n_fr_hr/__manifest__.py
Comment (quentinDupont) [reply l10n_fr_hr/__manifest__.py]: Depends on `hr`, hr.employee is a model in `hr`

Comment (legalsylvain) [reply l10n_fr_hr/__manifest__.py]: ok.

Comment (quentinDupont) [reply l10n_fr_hr/__manifest__.py]: nope.. `l10n_fr_department_hr_employee` can't work just depending on `hr` since we add the `country_of_birth` field in the `hr_cae` view..

What's the best way to do this ? 

Add in oca hr/hr.employee view
Then change it in `hr_cae` view making `hr_cae` depending on `l10n_fr_department_hr_employee` ?? :confused:

**Comment by robinkeunen** [comment]
Hey Quentin, sorry for the late reply

Go on with factorising the modules, it's a great idea.

Instead of just removing `base_location_geonames_import`, could you rather move it to `coopaname_custom`?

We need the module `account_banking_sepa_direct_debit` to export SEPA XML files, not just for the field `valid_mandate_id`.
If you think that's something only some CAE would want, we can add `account_banking_mandate` to `hr_cae_bank` and `account_banking_sepa_direct_debit` to `coopaname_custom` :-)

 The module names are cool for me.

What do you think @mclaeysb ?

File: l10n_fr_hr/__manifest__.py
Comment (robinkeunen) [reply l10n_fr_hr/__manifest__.py]: Mmh, if you're factorizing, it'd be a pity to add french dependencies into hr_cae. Maybe `hr_cae_fr`?

**Comment by quentinDupont** [comment]
> Hey Quentin, sorry for the late reply
> 
> Go on with factorising the modules, it's a great idea.
> 
> Instead of just removing `base_location_geonames_import`, could you rather move it to `coopaname_custom`?

Done :heavy_check_mark:

> 
> We need the module `account_banking_sepa_direct_debit` to export SEPA XML files, not just for the field `valid_mandate_id`.
> If you think that's something only some CAE would want, we can add `account_banking_mandate` to `hr_cae_bank` and `account_banking_sepa_direct_debit` to `coopaname_custom` :-)

I have no idea so I prefer not to change it.
> 
> The module names are cool for me.
> 
> What do you think @mclaeysb ?

So I rename in `hr_cae_fr` since we can't suggest to OCA since it depends on `hr_cae`

Actually. On second thought, many improvements are not dependent on a CAE at all. These are just improvement for HR, don't you think ?

hr_cae_recruitment →hr_recruitment ? 
hr_cae_bank → hr_bank
hr_cae_contribution for handling contributions :heavy_check_mark: it's really a CAE issue

**Comment by quentinDupont** [comment]
@robinkeunen I wait for your approval to rename modules name without `_cae_`
This PR is wide enough and prefer that we accept and merge it before continuing to factorize stuff :)

**Comment by robinkeunen** [comment]
- hr_cae_recruitment →hr_recruitment ?

I'd keep `hr_cae_recruitment`: `hr_recruitment` is too broad, if we were to rename this module following OCA modules, we'd need to split `hr_cae_recruitment` further into different functionnalities. For example: `hr_applicant_job`, `hr_applicant_certificate`, ...).

Plus there are things that are specific to CAE (equipment, turnover_minimum, ...). All in all, it's true that this module is totally specific to CAE but still, we are building a vertical solution for CAEs. 

- hr_cae_bank → hr_bank: Rather `hr_employee_bank` to be more specific and consisent with [OCA/hr](https://github.com/OCA/hr)

**Review by robinkeunen — APPROVED** [review]

**Review by mclaeysb — APPROVED** [review]
LGTM. I checked the resulting views for employee, and they were identical - nice work!

We're starting a new sprint just now. i'm merging this in.

Thanks a lot @quentinDupont !

**Comment by quentinDupont** [comment]
> LGTM. I checked the resulting views for employee, and they were identical - nice work!
> 
> We're starting a new sprint just now. i'm merging this in.
> 
> Thanks a lot @quentinDupont !

You're welcome, thanks !

**Pull request merged**
